presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcSelectivePageSourceFactory.java

```diff
@@ -360,23 +362,29 @@ public static OrcSelectivePageSource createOrcPageSource(
 
             List<FilterFunction> filterFunctions = toFilterFunctions(replaceExpression(remainingPredicate, variableToInput), bucketAdapter, session, rowExpressionService.getDeterminismEvaluator(), rowExpressionService.getPredicateCompiler());
 
-            OrcSelectiveRecordReader recordReader = reader.createSelectiveRecordReader(
-                    columnTypes,
-                    outputIndices,
-                    tupleDomainFilters,
-                    filterFunctions,
-                    inputs.inverse(),
-                    requiredSubfields,
-                    typedPrefilledValues,
-                    Maps.transformValues(mappedCoercers, Function.class::cast),
-                    orcPredicate,
-                    start,
-                    length,
-                    hiveStorageTimeZone,
-                    session.getSqlFunctionProperties().isLegacyMapSubscript(),
-                    systemMemoryUsage,
-                    Optional.empty(),
-                    INITIAL_BATCH_SIZE);
+            OrcSelectiveRecordReader recordReader;
+            try {
+                recordReader = reader.createSelectiveRecordReader(
+                        columnTypes,
+                        outputIndices,
+                        tupleDomainFilters,
+                        filterFunctions,
+                        inputs.inverse(),
+                        requiredSubfields,
+                        typedPrefilledValues,
+                        Maps.transformValues(mappedCoercers, Function.class::cast),
+                        orcPredicate,
+                        start,
+                        length,
+                        hiveStorageTimeZone,
+                        session.getSqlFunctionProperties().isLegacyMapSubscript(),
+                        systemMemoryUsage,
+                        Optional.empty(),
+                        INITIAL_BATCH_SIZE);
+            }
+            catch (NotSupportedException e) {
+                throw new PrestoException(NOT_SUPPORTED, e.getMessage(), e);
```

Comment:
Seems NOT_SUPPORTED is only thrown by OrcWriter but not reader. Did I miss anything?

Reply:
OrcSelectivePageSourceFactory.createSelectiveRecordReader() --> OrcReader.createSelectiveRecordReader() --> new OrcSelectiveRecordReader() --> super --> AbstractOrcRecordReader() --> createWriteStatisticsBuilder() --> new StatisticsValidation() --> new ColumnStatisticsValidation() --> NotSupportedException

Reply:
Got it. I think we can relax any orc writer validation-related callsites. Let's replace NOT_SUPPORTED with generic internal error for it. The validation code was introduced for orc writer rollout. It is not used anymore.

Reply:
Oh okay got it. Will throw as GenericInternalException.